### ISSUE TITLE
Question about difference between stop gradient and *.requires_grad_(False)

### ISSUE THREAD

**Issue by Kitsunetic** [threestudio-project/threestudio#297]
I think simply freezing the pretrained diffusion model using `*.requires_grad_(False)` is simpler to implement than stop gradient operation, and I think both act same.
So I'm not sure why the implementation in here are utilizing stop gradient operation instead of `requires_grad_`.
Is there any other reason or is my understanding wrong?
Can someone explain about this clearly?

**Comment by thuliu-yt16** [comment]
It has nothing to do with threestudio. 

But what did you mean by stop gradient? Is it `.detach()`? `.detach()` is used on intermediate variables to cut off the computation graph while `.requires_grad_` is called on parameters to disable recording gradients.
